### ISSUE TITLE
Double underscores in schema field names causing some grief

### ISSUE THREAD

**Issue by himynameisdave** [apollographql/eslint-plugin-graphql#92]
We have a few fields in our schema that start with a double underscore. One such example is a `__cost` field, which we use to enforce query limits. This causes an error with regards to it being named with a double underscore, and there doesn't appear to be any validation setting that can be passed in to turn this off.

**Error message:**
```
Name "__cost" must not begin with "__", which is reserved by GraphQL introspection. In a future release of graphql this will become a hard error.
```

**Relevant slice of our schema:**
```
...
{
    "name": "__cost",
    "description": null,
    "args": [],
    "type": {
        "kind": "SCALAR",
        "name": "JSONObject",
        "ofType": null
    },
    "isDeprecated": false,
    "deprecationReason": null
}
...
```

Would be nice to either turn this off or be able to pass a validation flag or something. It appears that the function where this error is originating from in the `graphql-js` package [already accepts an `isIntrospection`](https://github.com/graphql/graphql-js/blob/6e7d51d6c8fb9ed66d93510fa49477a05dd751af/src/utilities/assertValidName.js#L29) param that would disable this warning. However, [it does not appear that this is ever used](https://github.com/graphql/graphql-js/blob/4f5e351c2afb28e4ae3a44eae76d45d895ff3aa2/src/type/directives.js#L59) so a ticket may need to be filed in `graphql-js` as well.

**Comment by jnwng** [comment]
yeah, unfortunately we're just proxying validation errors from `graphql-js` itself, so we dont currently have a way to turn it off in `eslint-plugin-graphql`. i'd suggest you bring it upstream to `graphql-js`, and once `isIntrospection` is enabled we can add a flag for it here.
